addons/website_sale/models/website_snippet_filter.py

```diff
@@ -196,3 +205,16 @@ def _get_products_alternative_products(self, website, limit, domain, context):
                     display_default_code=False,
                 ).search(domain, limit=limit)
         return products
+
+    def _prepare_values(self, limit=None, search_domain=None):
+        hide_variants = False
+        if self.filter_id and search_domain:
+            if 'hide_variants' in search_domain:
+                hide_variants = True
+                search_domain.remove('hide_variants')
+            if not search_domain:
+                search_domain = None
```

Comment:
Yes because if search domain is `[]` then it will send that as the search domain of the `super()` call and it won't return any records, instead search domain is set as `None` so no domain is set on the records

Reply:
`search_domain = search_domain or None` then =]